modules/apps/foundation/roles/roles-item-selector-web/src/main/java/com/liferay/roles/item/selector/web/internal/RoleItemSelectorView.java

```diff
@@ -0,0 +1,128 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.roles.item.selector.web.internal;
+
+import com.liferay.item.selector.ItemSelectorReturnType;
+import com.liferay.item.selector.ItemSelectorView;
+import com.liferay.item.selector.criteria.UUIDItemSelectorReturnType;
+import com.liferay.portal.kernel.language.LanguageUtil;
+import com.liferay.portal.kernel.service.RoleLocalService;
+import com.liferay.portal.kernel.theme.ThemeDisplay;
+import com.liferay.portal.kernel.util.ListUtil;
+import com.liferay.portal.kernel.util.ResourceBundleUtil;
+import com.liferay.portal.kernel.util.WebKeys;
+import com.liferay.roles.item.selector.criterion.RoleItemSelectorCriterion;
+import com.liferay.roles.item.selector.web.internal.display.context.RoleItemSelectorViewDisplayContext;
+
+import java.io.IOException;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Locale;
+import java.util.ResourceBundle;
+
+import javax.portlet.PortletURL;
+
+import javax.servlet.RequestDispatcher;
+import javax.servlet.ServletContext;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Alessio Antonio Rendina
+ */
+@Component(immediate = true, service = ItemSelectorView.class)
+public class RoleItemSelectorView
+	implements ItemSelectorView<RoleItemSelectorCriterion> {
+
+	@Override
+	public Class<RoleItemSelectorCriterion> getItemSelectorCriterionClass() {
+		return RoleItemSelectorCriterion.class;
+	}
+
+	public ServletContext getServletContext() {
+		return _servletContext;
+	}
+
+	@Override
+	public List<ItemSelectorReturnType> getSupportedItemSelectorReturnTypes() {
+		return _supportedItemSelectorReturnTypes;
+	}
+
+	@Override
+	public String getTitle(Locale locale) {
+		ResourceBundle resourceBundle = ResourceBundleUtil.getBundle(
+			"content.Language", locale, getClass());
+
+		return LanguageUtil.get(resourceBundle, "roles");
+	}
+
+	@Override
+	public boolean isShowSearch() {
+		return true;
+	}
```

Comment:
Since the display context is not actually searching, do we want this to be true?  I suppose the use case will answer the question, but right now they don't match up.

Reply:
Hey @drewbrokke , you are right. It is better to add that kind of filtering that you're talking about. So feel free to change accordingly to Roles Admin portlet standard behaviour.